osbrain/tests/test_agent_ipc_sockets.py

```diff
@@ -0,0 +1,74 @@
+"""
+Test file for everything IPC socket related
+"""
+import pytest
+
+from osbrain import run_agent
+from osbrain import run_nameserver
+
+from common import nsproxy  # pragma: no flakes
+
+
+def test_agent_close_ipc_socket_agent_shutdown(nsproxy):
+    """
+    Check that the socket is closed and the socket file removed after the agent
+    is shut down.
+    """
+    agent = run_agent('name')
+    address = agent.bind('PUSH')
+    agent.shutdown()
+    assert not address.address.exists()
+
+
+def test_agent_close_ipc_socket_agent_kill(nsproxy):
+    """
+    Check that the socket is closed and the socket file removed after the agent
+    is killed.
+    """
+    agent = run_agent('name')
+    address = agent.bind('PUSH')
+    agent.kill()
+    assert not address.address.exists()
+
+
+def test_agent_close_ipc_socket_agent_blocked(nsproxy):
+    """
+    Check that the socket is closed and the socket file removed when a blocked
+    agent is requested to shut down.
+    """
+    def block(agent):
+        agent.send('out', 'blocking...')
+
+    blocker = run_agent('blocker')
+    blocker.set_method(block)
+    addr = blocker.bind('PUSH', alias='out')
+    blocker.after(0, 'block')
+    nsproxy.shutdown(timeout=1.)
+
+    assert not addr.address.exists()
+
+
+def test_agent_close_ipc_socket_agent_crash():
+    """
+    Check that the socket is closed and the socket file removed after the agent
+    crashes.
+    """
+    ns = run_nameserver()
+    agent = run_agent('agent')
+    addr = agent.bind('PUSH', 'main')
+    with pytest.raises(RuntimeError):
+        agent.raise_exception()
+
+    ns.shutdown()
+    assert not addr.address.exists()
+
+
+def test_agent_close_ipc_socket_nameserver_shutdown(nsproxy):
+    """
+    Check that the socket is closed and the socket file removed after the name
+    server is shut down.
+    """
+    agent = run_agent('agent')
+    addr = agent.bind('PUSH', 'main')
+    nsproxy.shutdown()
+    assert not addr.address.exists()
```

Comment:
Not very important but, for style consistency, I would rather keep the same spaces/blank lines in all these tests (i.e.: chose between leaving a blank space between the code and the assertions or not leaving one).

Reply:
I prefer leaving it. I'll check the other functions as well 👍